### PR TITLE
feat(licenseImport): fix and improve / add license admin portlet

### PR DESCRIPTION
**Pull request by maxhbr**
adds the functionallity to overwrite licenses with import when
- an external id matches uniqly or
- the license id matches

This adds also a new admin portlet, which collects all the license tasks to a single point. This makes a lar import necessary.

Additionally the import was broken and is now fixed again.

### CODE REVIEW

**Comment by maxhbr** [comment]
Screenshot of the new portlet
42631994-1d32424a-85dc-11e8-9f1f-3a4e72499ed6

**Comment by maierthomas** [comment]
here my test results:

- export of existing licenses OK
- import of existing licenses OK

-> changed some attributes of licenses and added the external id 'spdx-identifier'
-> reimport with parameters 'overwrite by correct external ids' OK

- delete all license information OK

- import ALL licenses from SPDX online FAILED
- import ALL licenses from SPDX offline/local FAILED
somehow its not possible for me to add SPDX licenses..

File: frontend/sw360-portlet/src/main/webapp/WEB-INF/liferay-portlet.xml [hunk 1]
```diff
@@ -67,6 +67,17 @@
         <header-portlet-javascript>/js/main.js</header-portlet-javascript>
     </portlet>
 
+    <portlet>
```
[the remark this answers is not in the document]
Comment (maxhbr) [reply frontend/sw360-portlet/src/main/webapp/WEB-INF/liferay-portlet.xml]: Maybe this .lar correctly excludes unnused parts like blogs, wikis, ...

File: backend/src/src-licenses/src/main/java/org/eclipse/sw360/licenses/tools/SpdxConnector.java [hunk 2]
```diff
@@ -54,7 +55,8 @@
                 .setShortname(spdxListedLicense.getLicenseId())
                 .setFullname(spdxListedLicense.getName())
                 .setText(spdxListedLicense.getLicenseText())
-                .setExternalLicenseLink("https://spdx.org/licenses/" + spdxListedLicense.getLicenseId()+ ".html");
+                .setExternalLicenseLink("https://spdx.org/licenses/" + spdxListedLicense.getLicenseId()+ ".html")
```
[the remark this answers is not in the document]
Comment (maxhbr) [reply backend/src/src-licenses/src/main/java/org/eclipse/sw360/licenses/tools/SpdxConnector.java]: This is the correct link corresponding to a SPDX license. I do not know what one would want to configure here.

File: frontend/sw360-portlet/src/main/webapp/html/admin/licenseAdmin/view.jsp [hunk 3]
```diff
@@ -0,0 +1,175 @@
+<%--
+  ~ Copyright Siemens AG, 2013-2017. Part of the SW360 Portal Project.
```
[the remark this answers is not in the document]
Comment (maxhbr) [reply frontend/sw360-portlet/src/main/webapp/html/admin/licenseAdmin/view.jsp]: Nearly all is copied and restructured code. But I will add my copyright to it.

**Comment by maxhbr** [comment]
@maierthomas I have tried to reproduce the problem with the SPDX import but was not able. Can you help me with explaining what your environment looked like?

Can you reproduce it and copy the stack trace?

**Comment by maxhbr** [comment]
This PR is now rebased and squashed

**Comment by maierthomas** [comment]
I just have this:

 2018-10-30 13:12:00 WARN  ListedLicenses:358 - Unable to access the SPDX listed licenses at http://www.spdx.org/licenses.  Using local file copy of SPDX listed licenses

File:
`ListedLicenses:358`

I am using the vagrant setup with proxy environment (cntlm).
SPDX uses jena rdf. May I need to configure the proxy settings here?
http://www.jaist.ac.jp/~hideaki/man/Jena/com/hp/hpl/jena/rdf/model/Model.html

`-DsocksProxyHost=[your-proxy-domain-name-or-ip-address] `
Well I need to configure my JVM?

**Comment by maxhbr** [comment]
It should work without an internet connection, otherwise it will not work on a standard deployment. But this is not the the topic of this PR and we should create an issue for that.

**Comment by maierthomas** [comment]
@maxhbr SPDX import works after adding proxy settings to the tomcat JVM like
`setenv.sh` (...-Dhttp.proxySet=true -Dhttp.proxyHost=<ip> -Dhttp.proxyPort=<port> ...)

**Comment by maxhbr** [comment]
interesting. I will have to investigate the behaviour without internet connection.

**Comment by maierthomas** [comment]
tested successfully.

File: frontend/sw360-portlet/src/main/webapp/WEB-INF/liferay-portlet.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (maierthomas) [reply frontend/sw360-portlet/src/main/webapp/WEB-INF/liferay-portlet.xml]: ok

[the remark this answers is not in the document]
Comment (maxhbr) [reply frontend/sw360-portlet/src/main/webapp/WEB-INF/liferay-portlet.xml]: Do you have tested it with import from scratch and import to existing instance?

**Pull request merged**